crypto/keys/ethsecp256k1/ethsecp256k1.go

```diff
@@ -0,0 +1,202 @@
+package ethsecp256k1
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/subtle"
+	"fmt"
+
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	tmcrypto "github.com/tendermint/tendermint/crypto"
+)
+
+const (
+	// PrivKeySize defines the size of the PrivKey bytes
+	PrivKeySize = 32
+	// PubKeySize defines the size of the PubKey bytes
+	PubKeySize = 33
+	// KeyType is the string constant for the EthSecp256k1 algorithm
+	KeyType = "ethsecp256k1"
+)
+
+// Amino encoding names
+const (
+	// PrivKeyName defines the amino encoding name for the EthSecp256k1 private key
+	PrivKeyName = "tendermint/PrivKeyEthSecp256k1"
+	// PubKeyName defines the amino encoding name for the EthSecp256k1 public key
+	PubKeyName = "tendermint/PubKeyEthSecp256k1"
+)
+
+// ----------------------------------------------------------------------------
+// secp256k1 Private Key
+
+var (
+	_ cryptotypes.PrivKey  = &PrivKey{}
+	_ codec.AminoMarshaler = &PrivKey{}
+)
+
+// GenerateKey generates a new random private key. It returns an error upon
+// failure.
+func GenerateKey() (*PrivKey, error) {
+	priv, err := ethcrypto.GenerateKey()
+	if err != nil {
+		return nil, err
+	}
+
+	return &PrivKey{
+		Key: ethcrypto.FromECDSA(priv),
+	}, nil
+}
+
+// Bytes returns the byte representation of the ECDSA Private Key.
+func (privKey PrivKey) Bytes() []byte {
+	return privKey.Key
+}
+
+// PubKey returns the ECDSA private key's public key.
+func (privKey PrivKey) PubKey() cryptotypes.PubKey {
+	ecdsaPrivKey := privKey.ToECDSA()
+	return &PubKey{
+		Key: ethcrypto.CompressPubkey(&ecdsaPrivKey.PublicKey),
+	}
+}
+
+// Equals returns true if two ECDSA private keys are equal and false otherwise.
+func (privKey PrivKey) Equals(other cryptotypes.LedgerPrivKey) bool {
```

Comment:
how do you compare the private key on the ledger against anything in software? Or is this `LedgerPrivKey` not the hardware device ledger meaning?